### ISSUE TITLE
Please add support for Ubuntu 18.10 and a Intel driver

### ISSUE THREAD

**Issue by Paul-dH** [AdnanHodzic/displaylink-debian#208]
Hi all,

First of all, thanks for providing a really easy to use installer for the DisplayLink driver :)

I do have some issues though, I did went trough all prerequisites and after that I installed the driver. After a reboot the service was active and there were no errors in dmesg.

The only thing is that my displays aren't recognized..

The debug command provided the info below:

```
--------------- Linux system info ----------------
Distro: Ubuntu
Release: cosmic
Kernel: 4.18.0-13-generic
---------------- DisplayLink info ----------------
Driver version: 4.4
EVDI service status: up and running
EVDI service version: 4.4.24
------------------ Graphics card -----------------
Vendor: i915
Subsystem: GT2
VGA: Intel Corporation Skylake GT2 [HD Graphics 520] (rev 07)
X11 version: 1.20.1-3ubuntu2.1
-------------- DisplayLink xorg.conf -------------
File: /etc/X11/xorg.conf.d/20-displaylink.conf
Contents:
 Section "Device"
    Identifier  "Intel"
    Driver      "intel"
EndSection
-------------------- Monitors --------------------
Providers: number : 1
Provider 0: id: 0x47 cap: 0x9, Source Output, Sink Offload crtcs: 4 outputs: 4 associated providers: 0 name:Intel
```

By the way, the hardware I'm using is a Fujitsu laptop with a USB3.0 Port Replicator (Fujitsu PR08).

**Comment by AdnanHodzic** [comment]
You're most welcome, please change contents of: `/etc/X11/xorg.conf.d/20-displaylink.conf` to following:

```
Section "OutputClass"
    Identifier  "DisplayLink"
    MatchDriver "evdi"
    Driver      "modesetting"
    Option      "AccelMethod" "none"
EndSection
```
After this logout/login (restart X) or simply restart the computer. Let me know on how it goes.

**Comment by Paul-dH** [comment]
Hi @AdnanHodzic,

I've changed the file and a debug output looks like this now:
```
--------------- Linux system info ----------------
Distro: Ubuntu
Release: cosmic
Kernel: 4.18.0-13-generic
---------------- DisplayLink info ----------------
Driver version: 4.4
EVDI service status: up and running
EVDI service version: 4.4.24
------------------ Graphics card -----------------
Vendor: i915
Subsystem: GT2
VGA: Intel Corporation Skylake GT2 [HD Graphics 520] (rev 07)
X11 version: 1.20.1-3ubuntu2.1
-------------- DisplayLink xorg.conf -------------
File: /etc/X11/xorg.conf.d/20-displaylink.conf
Contents:
#Section "Device"
#    Identifier  "Intel"
#    Driver      "intel"
#EndSection
Section "OutputClass"
    Identifier  "DisplayLink"
    MatchDriver "evdi"
    Driver      "modesetting"
    Option      "AccelMethod" "none"
EndSection
-------------------- Monitors --------------------
Providers: number : 1
Provider 0: id: 0x47 cap: 0x9, Source Output, Sink Offload crtcs: 4 outputs: 4 associated providers: 0 name:Intel
-------------------------------------------------------------------
```
I rebooted the laptop and tested the `xrandr --listproviders`, this only found the laptop screen like above. `gdm-session-worker` did report a crash though, so something somewhere did change.

Maybe another suggestion?

**Comment by AdnanHodzic** [comment]
And what happens if you make contents of `/etc/X11/xorg.conf.d/20-displaylink.conf` file to be:

```
Section "Device"
    Identifier  "DisplayLink"
    Driver      "modesetting"
    Option      "PageFlip" "false"
EndSection
```
Again followed by logout/login (restart X) or simply restart the computer. Let me know on how it goes.

**Comment by AdnanHodzic** [comment]
As part of https://github.com/AdnanHodzic/displaylink-debian/commit/b57cea89cc2e97de071e417bdd39ddd374072dbd commit udl/udlf will be added to blacklist.conf automatically if >= 4.14.9 is detected. 

I'm not sure if this will fix your problem, but I'd definitely advise you to pull the latest changes and perform reinstall and report back if there any changes.

**Comment by Paul-dH** [comment]
Update:
I did another re-install because the first line of the debug output showed the absence of the DisplayLink service. After a reboot I changed the config again to your last recommendation and both screens turned on again.

The issue of the black screens still persist, while I am able to get my pointer there. The list provider command also showed three screens now and Gnome Settings also finds the two displays.

**Comment by AdnanHodzic** [comment]
What do you mean by "last recommendation", what are the contents of `/etc/X11/xorg.conf.d/20-displaylink.conf` that's best working for you?

As from what I understand now is that you're displays are detected, but they are just "black" and you're able to get a pointer? You also consulted with [Post Install Guide](https://github.com/AdnanHodzic/displaylink-debian/blob/master/post-install-guide.md), right?

Also as part of https://github.com/AdnanHodzic/displaylink-debian/commit/b57cea89cc2e97de071e417bdd39ddd374072dbd commit udl/udlf will be added to blacklist.conf automatically if >= 4.14.9 is detected. 

I'm not sure if this will fix your problem, but I'd definitely advise you to pull the latest changes and perform reinstall and report back if there any changes.

**Comment by Paul-dH** [comment]
The content of `/etc/X11/xorg.conf.d/20-displaylink.conf` that produces the black screen with pointer is:

```
Section "Device"
    Identifier  "DisplayLink"
    Driver      "modesetting"
    Option      "PageFlip" "false"
EndSection
```

I did consult the PostInstall guide, almost double retried every step but I can't figure out why my screen stays black. I've also tries everything from issue #15 and #68 

With my previous post I already did a pull of the latest version and reinstalled the driver with the script :)

I've redone everything to the point that I have the black screens with the pointer and uses the latest version of your installer.

Now the monitors are being detected, their only just black.

```
Providers: number : 3
Provider 0: id: 0xa2 cap: 0x9, Source Output, Sink Offload crtcs: 4 outputs: 4 associated providers: 2 name:Intel
Provider 1: id: 0x7a cap: 0x2, Sink Output crtcs: 1 outputs: 1 associated providers: 1 name:modesetting
Provider 2: id: 0x41 cap: 0x2, Sink Output crtcs: 1 outputs: 1 associated providers: 1 name:modesetting

```

**Comment by AdnanHodzic** [comment]
I've just merged `displaylink-5` branch with `master` which introduces big changes, among which are support for yesterday released DisplayLink 5.1 driver. Disabling modesetting as part of `displaylink-debian` as now all of it should be happening on EVDI level. I've included few more changes.

Regardless, please pull latest (`git pull`) changes on `master` and run `sudo ./displaylink-debian.sh --install`

After this everything should be working as expected and with this said I'm closing the issue. If the issue still persists, re-open the issue and include output of `sudo ./displaylink-debian.sh --debug` command.
